### PR TITLE
Federated learning use case CI/CD

### PR DESCRIPTION
**Pull request by ferrarimarco**
Configure no-op cloud build triggers for the federated learning use case.

### CODE REVIEW

File: test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf [hunk 1]
```diff
@@ -219,3 +219,54 @@ resource "google_cloudbuild_trigger" "uc_mftp_model_eval_build" {
...
+  filename = "test/ci-cd/cloudbuild/uc-federated-learning-terraform.yaml"
+  ignored_files = [
+    "platforms/gke/base/core/README.md",
+    "platforms/gke/base/core/use-cases/README.md",
+  ]
+  included_files = [
+    # Include the whole core platform because we want to ensure that
+    # changes to the base platform don't break this use case
+    "platforms/gke/base/core/**",
+    "test/ci-cd/cloudbuild/uc-federated-learning-terraform.yaml",
```
Comment (arueth) [inline test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf:233]: Did you want to include `platforms/gke/base/use-cases/federated-learning/**`?

File: test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf [hunk 2]
```diff
@@ -219,3 +219,54 @@ resource "google_cloudbuild_trigger" "uc_mftp_model_eval_build" {
     }
   }
 }
+
+resource "google_cloudbuild_trigger" "uc_federated_learning_terraform" {
+  filename = "test/ci-cd/cloudbuild/uc-federated-learning-terraform.yaml"
+  ignored_files = [
+    "platforms/gke/base/core/README.md",
+    "platforms/gke/base/core/use-cases/README.md",
```
Comment (arueth) [inline test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf:227]: This should probably be `platforms/gke/base/use-cases/federated-learning/README.md`

Comment (ferrarimarco) [reply test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf:227]: Added both because there's `platforms/gke/base/core/use-cases/README.md` as well

Comment (arueth) [reply test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf:227]: That file wouldn't get picked up because it is not in any of the `included_files`.

Comment (ferrarimarco) [reply test/ci-cd/terraform/cloudbuild/cloudbuild_trigger.tf:227]: oh right, removing.

**Review by arueth — APPROVED** [review]

**Pull request merged**